modP2pImpl/src/org/aion/p2p/impl/comm/NodeMgr.java

```diff
@@ -168,20 +165,17 @@ public boolean isSeedIp(String _ip) {
     /** @param _n Node */
     @Override
     public void addTempNode(final INode _n) {
-        final ReentrantLock putLock = this.putLock;
+        tempNodesLock.lock();
         try {
-            putLock.lockInterruptibly();
-
             if (tempNodes.size() < maxTempNodes
                     && !tempNodes.containsKey(_n.getPeerId())
```

Comment:
Please update the JavaDoc to say that `_n` cannot be null.

Reply:
added null check and will return directly